src/network/webtask.cpp

```diff
@@ -123,32 +123,24 @@ void WebTask::onNetworkError(
         QNetworkReply::NetworkError errorCode,
         const QString& errorString,
         const WebResponseWithContent& responseWithContent) {
-    DEBUG_ASSERT(m_state == State::Pending);
+    DEBUG_ASSERT(m_state == State::Failed || m_state == State::Pending);
     DEBUG_ASSERT(m_timeoutTimerId == kInvalidTimerId);
 
     DEBUG_ASSERT(errorCode != QNetworkReply::NoError);
     switch (errorCode) {
-    case QNetworkReply::OperationCanceledError:
-        // Client-side abort or timeout
-        m_state = State::Aborted;
-        break;
-    case QNetworkReply::TimeoutError:
-        // Network or server-side timeout
+    case QNetworkReply::OperationCanceledError: // Client-side timeout
```

Comment:
Client- and server-side abort should be distinguished. Why did you merge them? The state `TimedOut` is wrong when manually aborting a request.

Reply:
We are here in onNetworkError() this is only called if a start request fails.
The case you mention is handled in slotAbort().
The change was required, because sometime MusicBrainz does not respond at all.
In that case the client Timothy of 60 s takes place. This case is a network error as well, and leads to a proper feedback about this situation to the user.

Reply:
https://github.com/mixxxdj/mixxx/pull/10884

Writing code is easier than fruitless arguing and explaining.

Override `doNetworkReplyAborted()` in the derived class and do whatever needs to be done.